pkg/cmd/provisioning/azure/create_oidc_issuer.go

```diff
@@ -0,0 +1,588 @@
+	ctxWithResp := runtime.WithCaptureResponse(context.Background(), &rawResponse)
+	getResourceGroupResp, err := client.ResourceGroupsClient.Get(
+		ctxWithResp,
+		resourceGroupName,
+		&armresources.ResourceGroupsClientGetOptions{})
+	if err != nil {
+		var respErr *azcore.ResponseError
+		if errors.As(err, &respErr) {
+			switch respErr.ErrorCode {
+			case "ResourceGroupNotFound":
+				// Resource group wasn't found so the resource group will need to be created
+				needToCreateResourceGroup = true
+			default:
+				return nil, errors.Wrapf(err, "unable to get resource group")
+			}
+		} else {
+			return nil, err
+		}
+	}
+
+	// Validate that existing resource group is in requested region
+	if getResourceGroupResp.Location != nil && *getResourceGroupResp.Location != region {
+		return nil, fmt.Errorf("found existing resource group %s in unexpected region=%s, requested region=%s",
+			*getResourceGroupResp.ResourceGroup.ID,
+			*getResourceGroupResp.Location,
+			region)
+	}
+
+	mergedResourceTags := map[string]*string{}
+	if getResourceGroupResp.Tags != nil {
+		for key, value := range getResourceGroupResp.Tags {
+			mergedResourceTags[key] = value
+		}
+	}
+
+	needToUpdateResourceGroup := false
+	for tagKey, tagValue := range resourceTags {
+		val, ok := mergedResourceTags[tagKey]
+		// Current tags don't contain expected tag keys and values
+		if !ok || (ok && *val != tagValue) {
+			mergedResourceTags[tagKey] = to.Ptr(tagValue)
+			needToUpdateResourceGroup = true
+		}
+	}
+
+	// Found and validated existing resource group, return
+	if !needToCreateResourceGroup && !needToUpdateResourceGroup {
+		log.Printf("Found existing resource group %s", *getResourceGroupResp.ResourceGroup.ID)
+		return &getResourceGroupResp.ResourceGroup, nil
+	}
+
+	// Create resource group
+	createResourceGroupResp, err := client.ResourceGroupsClient.CreateOrUpdate(
+		context.Background(),
+		resourceGroupName,
+		armresources.ResourceGroup{
+			Location: to.Ptr(region),
+			Tags:     mergedResourceTags,
+		},
+		nil)
+	if err != nil {
+		return nil, err
+	}
+	verb := "Updated"
+	if needToCreateResourceGroup {
+		verb = "Created"
+	}
+	log.Printf("%s resource group %s", verb, *createResourceGroupResp.ResourceGroup.ID)
+	return &createResourceGroupResp.ResourceGroup, nil
+}
+
+// ensureStorageAccount ensures that a storage account with storageAccountName exists within the provided resource group, region and subscription
+// resourceTags will be updated to match those provided to ensureStorageAccount if found to be different on an existing storage account.
+func ensureStorageAccount(client *azureclients.AzureClientWrapper, storageAccountName, resourceGroupName, region string, resourceTags map[string]string) (*armstorage.Account, error) {
+	listAccounts := client.StorageAccountClient.NewListByResourceGroupPager(resourceGroupName, &armstorage.AccountsClientListByResourceGroupOptions{})
+	list := make([]*armstorage.Account, 0)
+	for listAccounts.More() {
```

Comment:
Code review: could grab the existing storage account in this loop, eliminating the need to loop through them again below.